pkg/resource/config.go

```diff
@@ -184,54 +207,92 @@ func LoadAWSConfigFromAPIKeys(
 	token,
 	region,
 	roleArn,
+	roleSessionName,
 	externalId string,
 ) (*aws.Config, error) {
-	configOptions := []func(*config.LoadOptions) error{
-		config.WithRegion(region),
-		config.WithCredentialsProvider(
-			credentials.NewStaticCredentialsProvider(
-				accessKeyID,
-				secretAccessKey,
-				token,
+	configOptions := []func(*config.LoadOptions) error{}
+
+	// use specified region if provided
+	if region != "" {
+		configOptions = append(
+			configOptions,
+			config.WithRegion(region),
+		)
+	}
+
+	// use static credentials if provided
+	if accessKeyID != "" && secretAccessKey != "" {
+		configOptions = append(
+			configOptions,
+			config.WithCredentialsProvider(
+				credentials.NewStaticCredentialsProvider(
+					accessKeyID,
+					secretAccessKey,
+					token,
+				),
```

Comment:
Allow these configuration functions to be appended to `configOptions` conditionally.